training-tracker-service/src/main/java/org/tolking/training_event_service/service/impl/TrainingEventServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package org.tolking.training_event_service.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+import org.tolking.training_event_service.dto.TrainingEventDTO;
+import org.tolking.training_event_service.entity.TrainingEvent;
+import org.tolking.training_event_service.enums.ActionType;
+import org.tolking.training_event_service.model.TrainerSummary;
+import org.tolking.training_event_service.repository.TrainerRepository;
+
+import java.time.LocalDate;
+import java.time.YearMonth;
+import java.time.ZoneId;
+import java.util.*;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class TrainingEventServiceImpl implements org.tolking.training_event_service.service.TrainingEventService {
+    private final TrainerRepository trainerRepository;
+    private final ModelMapper modelMapper;
+
+    @Override
+    public void create(TrainingEventDTO trainingEventDTO){
+        log.debug("Creating event with DTO: {} ", trainingEventDTO);
+        TrainingEvent trainingEvent = modelMapper.map(trainingEventDTO, TrainingEvent.class);
+
+        TrainingEvent createdEvent = trainerRepository.save(trainingEvent);
```

Comment:
I think both approaches is ok, but since this service just calculates durations per period and we are not interested in actual events, just its agregates, better to store only final aggregates instead of events. It'll help to reduce disk consumtion and performance of `getAll` will be much better

Reply:
Could you please share requirement? As I see we asked only to store statistics
![image](https://github.com/user-attachments/assets/891dfa0e-0ed0-4db0-b50d-d9aaf40dae34)

Reply:
![image](https://github.com/user-attachments/assets/1e451db9-9bef-4819-aac4-0b81bbbc58bb)
Hmm, I thought, with this rest endpoint we have to save data and calculate summary according to it. 

Do you think that we have to save only summary in in-memory db?

Reply:
From what I see, bullet 2 defines API specification, bullet 3 defines what should be done as result of it.
I'm not speaking it explicelty says "do not save events", but there's no such requirement. We could save on summary to decrease disk usage and increase performance of /summary endpoint